server/src/main/java/org/opensearch/cluster/routing/ShardRouting.java

```diff
@@ -75,6 +75,7 @@ public class ShardRouting implements Writeable, ToXContentObject {
     private final long expectedShardSize;
     @Nullable
     private final ShardRouting targetRelocatingShard;
+    private boolean assignedToRemoteStoreNode;
```

Comment:
Ideally we don't want to serialize the value for this field since this won't be used anywhere else. I have changed this to `Boolean` instead of the primitive type, but I have assigned a default value to prevent any possible NPEs when the shardRouting entry sent from the cluster manager nodes are serialized on the data node.